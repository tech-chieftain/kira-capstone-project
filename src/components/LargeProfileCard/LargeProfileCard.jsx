import PropTypes from "prop-types";

import Link from "next/link";

import Card from "react-bootstrap/Card";
import Row from "react-bootstrap/Row";
import Col from "react-bootstrap/Col";
import Image from "react-bootstrap/Image";
import Button from "react-bootstrap/Button";

import { FaUserCircle } from "react-icons/fa";

const LargeProfileCard = ({ uid, photoURL, displayName, job, location, about, skills }) => (
  <Card className="shadow">
    <Card.Body>
      <Row className="align-items-center">
        <Col xs={4}>
          <Link href="/profile/[]" as={`/profile/${uid}`}>
            <a>
              {photoURL ? (
                <Image src={photoURL} alt="" roundedCircle fluid />
              ) : (
                <div className="d-flex justify-content-end">
                  <FaUserCircle size="4em" />
                </div>
              )}
            </a>
          </Link>
        </Col>
        <Col>
          <Link href="/profile/[]" as={`/profile/${uid}`}>
            <a className="text-decoration-none text-reset">
              <Card.Title className="text-primary">{displayName}</Card.Title>
            </a>
          </Link>
          <Link href={`/freelancers_search?q=${job}`}>
            <a className="text-decoration-none text-reset">
              <Card.Subtitle>{job}</Card.Subtitle>
            </a>
          </Link>
          <Link href={`/freelancers_search?q=${location}`}>
            <a className="text-decoration-none text-reset">
              <Card.Text className="fw-light">{location}</Card.Text>
            </a>
          </Link>
        </Col>
      </Row>
      <Row>
        <Col className="my-3">
          <Card.Text>{about}</Card.Text>
        </Col>
      </Row>
      <Row>
        <Col>
          {skills.slice(0, 5).map((tag) => (
            <Link href={`/freelancers_search?q=${tag}`}>
              <a className="text-decoration-none text-reset">
                <Button variant="outline-primary me-2 mb-2">{tag}</Button>
              </a>
            </Link>
          ))}
        </Col>
      </Row>
    </Card.Body>
  </Card>
);

LargeProfileCard.propTypes = {
  uid: PropTypes.string.isRequired,
  photoURL: PropTypes.string,
  displayName: PropTypes.string.isRequired,
  job: PropTypes.string,
  location: PropTypes.string,
  about: PropTypes.string,
  skills: PropTypes.array,
};

LargeProfileCard.defaultProps = {
  photoURL: "",
  job: "",
  location: "",
  about: "",
  skills: [],
};

export default LargeProfileCard;
