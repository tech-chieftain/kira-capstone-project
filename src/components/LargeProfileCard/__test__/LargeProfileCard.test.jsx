import * as React from "react";
import * as renderer from "react-test-renderer";
import LargeProfileCard from "../LargeProfileCard";

it("renders LargeProfileCard with correctly", () => {
  const tree = renderer
    .create(
      <LargeProfileCard
        displayName="John Doe"
        uid="12345"
        username="john"
        name="John Doe"
        profilePicture="https://picsum.photos/100/100"
        job="Web Developer"
        location="Baghdad"
        about="Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua."
        tags={["Front-End Development", "Website", "Wordpress", "PWA"]}
        topRated
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
