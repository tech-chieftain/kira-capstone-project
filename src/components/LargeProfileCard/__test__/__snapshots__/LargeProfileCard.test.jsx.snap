// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders LargeProfileCard with correctly 1`] = `
<div
  className="shadow card"
  style={
    Object {
      "width": "18rem",
    }
  }
>
  <div
    className="card-body"
  >
    <div
      className="align-items-center row"
    >
      <div
        className="col-4"
      >
        <a
          href="/user/john"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          <img
            alt=""
            className="img-fluid rounded-circle"
            src="https://picsum.photos/100/100"
          />
        </a>
      </div>
      <div
        className="col"
      >
        <a
          className="text-decoration-none text-reset"
          href="/user/john"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          <div
            className="text-primary card-title h5"
          >
            John Doe
          </div>
        </a>
        <a
          className="text-decoration-none text-reset"
          href="/services_search?q=Web%20Developer"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          <div
            className="card-subtitle h6"
          >
            Web Developer
          </div>
        </a>
        <p
          className="fw-light card-text"
        >
          Baghdad
        </p>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="my-3 col"
      >
        <p
          className="card-text"
        >
          Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
        </p>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col"
      >
        <a
          className="text-decoration-none text-reset"
          href="/services_search?q=Front-End%20Development"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          <button
            className="btn btn-outline-primary me-2 mb-2"
            disabled={false}
            type="button"
          >
            Front-End Development
          </button>
        </a>
        <a
          className="text-decoration-none text-reset"
          href="/services_search?q=Website"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          <button
            className="btn btn-outline-primary me-2 mb-2"
            disabled={false}
            type="button"
          >
            Website
          </button>
        </a>
        <a
          className="text-decoration-none text-reset"
          href="/services_search?q=Wordpress"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          <button
            className="btn btn-outline-primary me-2 mb-2"
            disabled={false}
            type="button"
          >
            Wordpress
          </button>
        </a>
        <a
          className="text-decoration-none text-reset"
          href="/services_search?q=PWA"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          <button
            className="btn btn-outline-primary me-2 mb-2"
            disabled={false}
            type="button"
          >
            PWA
          </button>
        </a>
      </div>
    </div>
    <hr />
    <div
      className="row"
    >
      <div
        className="text-end fluid col"
      >
        <p
          className="d-flex card-text"
        >
          <svg
            className="align-self-center m-1"
            fill="currentColor"
            height="1rem"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 576 512"
            width="1rem"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
            />
          </svg>
           Top Rated
        </p>
      </div>
    </div>
  </div>
</div>
`;
